### PR TITLE
Use allele info in VariantContext comparisons for stable sorts

### PR DESCRIPTION
**Pull request by clintval**
TLDR: This PR will make coordinate VCF sorting in HTSJDK/Picard similar to bcftools.

I tried to use Picard's [SortVcf](https://gatk.broadinstitute.org/hc/en-us/articles/360036858331-SortVcf-Picard) for a VCF comparison task, but I had difficulty because it's output was non-stable based on the full primary information of a variant context (contig, start, and alleles). I finally chose the [`bcftools`](https://samtools.github.io/bcftools/bcftools.html#sort) implementation because it [has a slightly more-stable sorting method](https://github.com/samtools/bcftools/blob/dd937aa89b3998aa4c6fd15375206e19c22d3f20/vcfsort.c#L95-L117). However, I want to ensure future users of Picard don't hit the same issue I did, hence this PR.

Before I write unit tests, is this something you are willing to accept?

### CODE REVIEW

**Comment by clintval** [comment]
Any interest from the maintainers in this PR?

Anything I can do to help move it closer to merge?

**Comment by clintval** [comment]
@cmnbroad I hit another instance where this would have helped me.

Do you want me to invest the time to implement some unit tests, or is there no interest in pursuing this PR?

**Comment by cmnbroad** [comment]
@clintval Stable sounds good to me, but @lbergelson really needs to weigh in (I'm not actually a maintainer).

**Comment by lbergelson** [comment]
@clintval I'm sorry I haven't been responsive.  This is a good idea and having some tests would be great.

**Comment by clintval** [comment]
@lbergelson I don't normally write in Java so I hope my last commits are OK!

**Review by lbergelson — APPROVED** [review]
@clintval looks good to me!  Thank you.  I have a few minor things but I'll take them over and push my own patch since I've left you sitting for so long.

**Comment by clintval** [comment]
@lbergelson nice work and thank you kindly for the review!
